test/langtools/tools/javac/parser/ExtraPackageSemicolon.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8305671
+ * @summary Verify an extra semicolon is allowed after package decl with no imports
+ * @modules jdk.compiler/com.sun.tools.javac.code
+*/
+
+import com.sun.tools.javac.Main;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+public class ExtraPackageSemicolon {
+
+    public static void runTest(String filename, String source) throws Exception {
+        final File sourceFile = new File(filename);
+        System.err.println("writing: " + sourceFile);
+        try (PrintStream output = new PrintStream(new FileOutputStream(sourceFile))) {
+            output.println(source);
```

Comment:
This would be somewhat overly simplistic (but not wrong) if any source had newlines in it.

Reply:
Fixed by using `ToolBox`.